app/helpers/application_cocoon_helper.rb

```diff
@@ -0,0 +1,53 @@
+module ApplicationCocoonHelper
+  # Shows a link that will allow to dynamically add a new associated object. This is a wrapper of
+  #   {#link_to_add_association}
+  #
+  # @overload link_to_association(name, form, options, association, html_options = nil)
+  #   Shows a link that will allow to dynamically add a new associated object.
+  #   @param [String] name the text to show in the link.
+  #   @param [FormBuilder] form the form builder.
+  #   @param [Symbol] association the associated objects, this should be the name of the has_many
+  #   relation.
+  #   @param [Hash] html_options the html options to be passed to {#link_to}.
+  #   @option html_options [Symbol] :anchor_node the jquery selector of the node. Example
+  #     values: `before`, `after`, `append`, `prepend`, etc. Default: parent node.
+  #   @option html_options [Symbol] :insertion_method ('before') jquery method that inserts the
+  #     new data. Example values: `before`, `after`, `append`, `prepend`, etc.
+  #   @option html_options [Symbol] :traversal the jquery traversal method to allow node selection
+  #     relative to the link. Example values: `closest`, `next`, `children`, etc. Default: absolute
+  #     selection.
+  # @overload link_to_association(form, options, association, html_options = nil, &block)
+  #   Shows a link that will allow to dynamically add a new associated object. The display of the
+  #   link can be changed in the block.
+  #   @param [String] name the text to show in the link.
+  #   @param [FormBuilder] form the form builder.
+  #   @param [Symbol] association the associated objects, this should be the name of the has_many
+  #   relation.
+  #   @param [Hash] html_options the html options to be passed to {#link_to}
+  #   @option html_options [Symbol] :anchor_node the jquery selector of the node. Example
+  #     values: `before`, `after`, `append`, `prepend`, etc. Default: parent node.
+  #   @option html_options [Symbol] :insertion_method ('before') jquery method that inserts the
+  #     new data. Example values: `before`, `after`, `append`, `prepend`, etc.
+  #   @option html_options [Symbol] :traversal the jquery traversal method to allow node selection
+  #     relative to the link. Example values: `closest`, `next`, `children`, etc. Default: absolute
+  #     selection.
+  #   @param [Proc] block The block to use for displaying the link.
+  def link_to_association(name, form, association = nil, html_options = nil, &block)
+    name, form, association, html_options = nil, name, form, association if block_given?
+    html_options = html_options.try(:dup) || {}
+    replace_keys(html_options)
+    link_to_add_association(*[name, form, association, html_options].compact, &block)
+  end
+
+  private
+
+  def replace_keys(html_options)
+    anchor_node = html_options.delete(:anchor_node)
+    insertion_method = html_options.delete(:insertion_method)
+    traversal = html_options.delete(:traversal)
```

Comment:
The new node will be added to the place where `link_to` is, this allows you to specify the node selection method, so that new node's place can be customised. e.g. `next` or `closet` to `link_to`